widgets/src/components/SummaryCard/SummaryCardContainer.js

```diff
@@ -0,0 +1,211 @@
+
+    this.state = {
+      config: null,
+      loading: false,
+      loadingValues: false,
+      summary: null,
+      period: 'monthly',
+    };
+
+    this.fetchSummary = this.fetchSummary.bind(this);
+    this.fetchWidgetConfigs = this.fetchWidgetConfigs.bind(this);
+    this.handlePeriodChange = this.handlePeriodChange.bind(this);
+  }
+
+  componentDidMount() {
+    const { serviceUrl } = this.props;
+    if (!LOCAL) {
+      // set the PDA domain to the URL passed via props
+      DOMAINS.PDA = serviceUrl;
+    }
+
+    this.setState({ loading: true, loadingValues: true }, async () => {
+      const { config: storedConfig } = this.state;
+      const config = storedConfig || (await this.fetchWidgetConfigs());
+
+      this.setState({ config }, () => this.fetchSummary());
+    });
+  }
+
+  async fetchWidgetConfigs() {
+    const { pageCode, frameId } = this.props;
+    try {
+      // config will be fetched from app-builder
+      const widgetConfigs = await getPageWidget(pageCode, frameId);
+      if (widgetConfigs.errors && widgetConfigs.errors.length) {
+        throw widgetConfigs.errors[0];
+      }
+
+      const { config } = widgetConfigs.payload;
+
+      return config;
+    } catch (error) {
+      this.handleError(error.message);
+    }
+    return null;
+  }
+
+  async fetchSummary() {
+    const { config, period } = this.state;
+
+    const connection = (config && config.knowledgeSource) || '';
+    const settings = (config && config.settings && JSON.parse(config.settings)) || '';
+
+    try {
+      const summary = await getSummary(connection, settings.summaryId, period);
+
+      this.setState({
+        loading: false,
+        loadingValues: false,
+        summary: (summary && summary.payload) || null,
+      });
+    } catch (error) {
+      this.handleError(error.message);
+    }
+  }
+
+  handleError(err) {
+    const { onError } = this.props;
+    onError(err);
+  }
+
+  handlePeriodChange(event) {
+    this.setState({ period: event.target.value, loadingValues: true }, () => this.fetchSummary());
+  }
+
+  renderSkeletonHeader() {
+    const { classes } = this.props;
+    return (
+      <div className={classes.header}>
+        <Skeleton width="100px" height={27} variant="rect" />
+        <Skeleton width="60px" height={27} variant="rect" />
+      </div>
+    );
+  }
+
+  renderHeader() {
+    const { period, summary } = this.state;
+    const { classes } = this.props;
+    return (
+      <div className={classes.header}>
+        <Typography variant="subtitle2" component="h3" className={classes.headline}>
+          {i18next.t(`card.labels.${summary.title}`)}
+        </Typography>
+        <Select
+          value={period}
+          variant="outlined"
+          classes={{
+            root: classes.periodSelectRoot,
+            iconOutlined: classes.periodSelectIcon,
+          }}
+          onChange={this.handlePeriodChange}
+        >
+          {['monthly', 'annually', 'daily'].map(periodi => (
```

Comment:
what is `periodi`?

Reply:
I didn't think too much on naming after I found out I can't name it as period, but nevertheless, I changed it to `periodItem`